clapi-hs.cabal

```diff
@@ -52,6 +52,7 @@ library
                      , Clapi.Types.Definitions
                      , Clapi.Types.Digests
                      , Clapi.Types.Dkmap
+                     , Clapi.Types.Nat
```

Comment:
Went down the rabbit hole on this a bit and found: http://hackage.haskell.org/package/typelits-witnesses
Which is sort of deprecated in favour of the singletons library.
Not saying we should go there, but there's some prior art for making proofs with TypeLits (if I understood correctly).

Reply:
Yeah, I'd had a look at that. I think it still relies on using type-level functions that result in type-level bools, rather than inhabited types. So, for example, the `a <= b` typeclass/constraint thing actually under the hood gave you an `a :<=: b ~ 'True` constraint to work with. When I asked about it on IRC and read some other stuff, I got the impression that the booleans didn't get you that far, whereas the inhabited GADTs where more useful.

I think probably `singletons` is where we'll end up if we change this again before dependantly typed Haskell stuff starts landing in GHC around 2020, but I needed to go through it at least a little bit "dumb" for myself, just so I could get my head around what was going on!